packages/strapi-mongoose/lib/relations.js

```diff
@@ -0,0 +1,353 @@
+
+              // Push the work into the flow process.
+              toAdd.forEach(value => {
+                value = _.isString(value) ? { [this.primaryKey]: value } : value;
+
+                if (association.nature === 'manyToMany' && !_.isArray(params.values[this.primaryKey] || params[this.primaryKey])) {
+                  value[details.via] = (value[details.via] || [])
+                    .concat([(params.values[this.primaryKey] || params[this.primaryKey])])
+                    .filter(x => {
+                      return x !== null && x !== undefined;
+                    });
+                } else {
+                  value[details.via] = params[this.primaryKey] || params.id;
+                }
+
+                virtualFields.push(
+                  module.exports.addRelation.call(model, {
+                    id: value[this.primaryKey] || value.id || value._id,
+                    values: _.pick(value, [this.primaryKey, details.via]),
+                    foreignKey: current
+                  })
+                );
+              });
+
+              toRemove.forEach(value => {
+                value = _.isString(value) ? { [this.primaryKey]: value } : value;
+
+                if (association.nature === 'manyToMany' && !_.isArray(params.values[this.primaryKey])) {
+                  value[details.via] = value[details.via].filter(x => x.toString() !== params.values[this.primaryKey].toString());
+                } else {
+                  value[details.via] = null;
+                }
+
+                virtualFields.push(
+                  module.exports.removeRelation.call(model, {
+                    id: value[this.primaryKey] || value.id || value._id,
+                    values: _.pick(value, [this.primaryKey, details.via]),
+                    foreignKey: current
+                  })
+                );
+              });
+            } else if (_.get(this._attributes, `${current}.isVirtual`) !== true) {
+              acc[current] = params.values[current];
+            }
+
+            break;
+          case 'manyMorphToMany':
+          case 'manyMorphToOne':
+            // Update the relational array.
+            acc[current] = params.values[current].map(obj => {
+              const globalId = obj.source && obj.source !== 'content-manager' ?
+                strapi.plugins[obj.source].models[_.toLower(obj.ref)].globalId:
+                strapi.models[_.toLower(obj.ref)].globalId;
+
+              // Define the object stored in database.
+              // The shape is this object is defined by the strapi-mongoose connector.
+              return {
+                ref: obj.refId,
+                kind: globalId,
+                [association.filter]: obj.field
+              }
+            });
+            break;
+          case 'oneToManyMorph':
+          case 'manyToManyMorph':
+            const transformToArrayID = (array) => {
+              if (_.isArray(array)) {
```

Comment:
It doesn't work when the `entry` is null because `_.castArray()` will return `[null]`, and it will break during the `map(id => id.toString())`. We need to check manually the value of `entry` and return an empty array instead.